pkg/deployment/resources/pod_cleanup.go

```diff
@@ -55,15 +52,19 @@ func (r *Resources) CleanupTerminatedPods(ctx context.Context, cachedStatus insp
 			return nil
 		}
 
-		if !(k8sutil.IsPodSucceeded(pod) || k8sutil.IsPodFailed(pod) || k8sutil.IsPodTerminating(pod)) {
```

Comment:
@ajanikow I moved this code below, so in consequence it will not be launched when it is not possible to `MemberStatusByPodName` in the function below.

Reply:
It is fine